exp-models/addon/models/experiment.js

```diff
@@ -21,14 +26,32 @@ export default DS.Model.extend(JamModel, {
         return Ember.isEqual(this.get('active'), 'Active');
     }),
 
-    administrators: DS.hasMany('admin'),
-    history: DS.hasMany('history'),
-    sessions: DS.hasMany('session'),
-
     eligibilityCriteria: DS.attr('string'),
     eligibilityString: Ember.computed('eligibilityCriteria', function() {
         var eligibility = this.get('eligibilityCriteria');
         // TODO
         return eligibility || "None";
-    })
+    }),
+
+    history: DS.hasMany('history'),
+
+    sessionCollectionId: Ember.computed('shortId', function() {
+        // Return a string corresponding to the session collection shortID, to be used by model/adapter/serializer
+        // Eg an experiment called 'test0' would have a collection 'session-test0'
+        return `session${this.get('shortId')}s`; // FIXME: the spurious s is a requirement imposed by genschemas...
+    }),
+
+    _registerSessionModels() {
+        // Dynamically register the required models for a session table associated with this experiment
+        var cId = this.get('sessionCollectionId');
+        window.App.register(`model:${cId}`, SessionModel.extend()); // register a dummy model. This seems to work even if model already registered
```

Comment:
It makes me a little worried too. 

It seems that Ember [removed](http://emberigniter.com/accessing-global-object-in-ember-cli-app/) the global app reference it used to have, and this approach seems to depend on calling `application.register`... I honestly don't know if there's another way to get at the app instance.

This is about all I could find on the lifetime of the window object, but that's a thin shield. I agree I'd like to see a better way.